task_3/main.cpp

```diff
@@ -0,0 +1,86 @@
+#include "counter.h"
+
+#include <iostream>
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+#include <boost/thread/thread.hpp>
+#include <experimental/filesystem>
+#include <chrono>
+#include <queue>
+#include <fstream>
+#include <string>
+
+using std::string;
+using std::cout;
+using std::endl;
+using namespace std::experimental::filesystem;
+
+extern boost::mutex global_stream_lock;
+void WorkerThread( boost::shared_ptr< boost::asio::io_service > io_service );
+
+int main( int argc, char * argv[] ) {
+	path dir {argc > 1 ? argv[1] : current_path()};				//	getting initial path
```

Comment:
Program can ask user to use the current directory if no arguments are present. This behavior is hidden so user should read full documentation. In future, while making UI, think about clarity of the input format for the user. For example even after reading the code it is unclear what directory format will be applied "C:\\dir" or C:\\dir or C:/dir etc.

Reply:
agree with it